File: code_snippets/python/src/guides/sagas/clients/flight_client.py
import logging

from pydantic import BaseModel

logger = logging.getLogger(__name__)


class FlightRequest(BaseModel):
    flight_id: str
    passenger_name: str


async def reserve(customer_id: str, request: FlightRequest):
    # this should implement the communication with the flight
    # provider's APIs
    logging.info(f"Flight reservation cancelled for customer: {customer_id}")
    return "my id"


async def cancel(booking_id: str):
    # this should implement the communication with the flight
    # provider's APIs
    logging.info(f"Flight reservation cancelled for customer: {booking_id}")


async def confirm(booking_id: str):
    # this should implement the communication with the flight
    # provider's APIs
    logging.info(f"Flight reservation cancelled for customer: {booking_id}")

File: code_snippets/ts/src/guides/sagas/clients/flight_client.ts
export const flightClient = {
  reserve: async (
    customerId: string,
    _req: { flightId: string; passengerName: string }
  ) => {
    console.info(`Flight booked for customer ${customerId}`);
    return "my id";
  },
  cancel: async (bookingId: string) => {
    console.info(`Flight cancelled for customer ${bookingId}`);
  },
  confirm: async (bookingId: string) => {
    console.info(`Flight confirmed for customer ${bookingId}`);
  },
};

export const paymentClient = {
  charge: async (paymentInfo: string, paymentId: string) => {},

  refund: async (paymentId: string) => {},
};
